internal/middlewares/authelia_context.go

```diff
@@ -107,6 +113,24 @@ func (c *AutheliaCtx) XForwardedURI() []byte {
 	return c.RequestCtx.Request.Header.Peek(xForwardedURIHeader)
 }
 
+// ForwardedProtoHost gets the X-Forwarded-Proto and X-Forwarded-Host headers and forms them into a URL.
+func (c AutheliaCtx) ForwardedProtoHost() (string, error) {
+	XForwardedProto := c.XForwardedProto()
+
+	if XForwardedProto == nil {
+		return "", errMissingXForwardedProto
+	}
+
+	XForwardedHost := c.XForwardedHost()
+
+	if XForwardedHost == nil {
+		return "", errMissingXForwardedHost
+	}
```

Comment:
This would account for Traefik and OIDC:

```suggestion
	XForwardedHost := c.XForwardedHost()

	XForwardedHost := c.XForwardedHost()

	if XForwardedHost == nil {
		XForwardedHost = c.Host()
	}
	
	if XForwardedHost == nil {
		return "", errMissingXForwardedHost
	}
```